internal/servers/controller/handlers/host_sets/host_set_service.go

```diff
@@ -850,17 +866,23 @@ func validateCreateRequest(req *pbs.CreateHostSetRequest) error {
 	})
 }
 
-func validateUpdateRequest(req *pbs.UpdateHostSetRequest) error {
+func validateUpdateRequest(ctx context.Context, req *pbs.UpdateHostSetRequest) error {
 	return handlers.ValidateUpdateRequest(req, req.GetItem(), func() map[string]string {
 		badFields := map[string]string{}
+		if len(req.GetItem().GetPreferredEndpoints()) > 0 {
+			_, err := endpoint.NewPreferencer(ctx, endpoint.WithPreferenceOrder(req.GetItem().GetPreferredEndpoints()))
+			if err != nil {
+				badFields[globals.PreferredEndpointsField] = fmt.Errorf("Error parsing preferred endpoints: %w.", err).Error()
+			}
+		}
 		switch host.SubtypeFromId(req.GetId()) {
 		case static.Subtype:
 			if req.GetItem().GetType() != "" && req.GetItem().GetType() != static.Subtype.String() {
-				badFields["type"] = "Cannot modify the resource type."
+				badFields[globals.TypeField] = "Cannot modify the resource type."
 			}
 		}
 		return badFields
-	}, static.HostSetPrefix)
+	}, static.HostSetPrefix, plugin.HostSetPrefix)
```

Comment:
This won't end up working when we support plugin host sets unless we do not have plugin ids with 2 `_`s.  The check for valid id currently checks for a single `_` after the prefix and only alphanumeric values following that.

Reply:
Agreed, however, since we haven't decided on the final format...and if that final format only has one underscore it will work fine. 🤷